mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -123,7 +123,19 @@ def sanity_check_ast(self) -> None:
             raise InvalidCode('No statements in code.')
         first = self.ast.lines[0]
         if not isinstance(first, mparser.FunctionNode) or first.func_name != 'project':
-            raise InvalidCode('First statement must be a call to project')
+            p = pathlib.Path(self.source_root).resolve()
+            found = p
+            for parent in p.parents:
+                if (parent / 'meson.build').is_file():
+                    found = parent
+                else:
+                    break
```

Comment:
All subdirs don't necessarily contain `meson.build` because you can do `subdir('foo/bar')`.

Reply:
I noted this in the commit message, meson will try to provide a guess but isn't meant to be perfect.